srtcore/core.cpp

```diff
@@ -8952,6 +8983,20 @@ int CUDT::packLostData(CPacket& w_packet, steady_clock::time_point& w_origintime
             continue;
         }
 
+        if (m_bPeerNakReport && m_iOPT_RexmitAlgo != 0)
+        {
+            steady_clock::time_point tsLastRexmit;
+            m_pSndBuffer->getPacketTime(offset, tsLastRexmit);
```

Comment:
I think here is the problem: you are notifying the last retransmission time of particular packet in the packet data, so in order to dig up for it, you have to run a suboptimal procedure to find the packet in a structure that was not intended for this searching, just to find the retransmission time. If you instead search for the sequence number if this packet in the sender loss list, you'll find it faster, as well as this opens a future possibility to do uncyclic NAKreports.

Reply:
Ok, the same problem is when getting the packet for retransmission (`CSndBuffer::readData` with offset). Forget it.